handshake_messages.go

```diff
@@ -148,36 +126,14 @@ func (m *clientHelloMsg) marshal() ([]byte, error) {
 			})
 		})
 	}
-	if m.ticketSupported {
+	if m.ticketSupported && !echInner {
```

Comment:
We may want to provide full control for the user over the outer client hello, even if they're providing an ECH inner. E.g. maybe they want a ticketSupported extension in the outer client hello, and in the inner?

Reply:
Maybe ignore this comment :)

Reply:
Good point for the ech PR, we will want to provide some way for the user to control wether an extension should be included in the ECH inner or not.